tensorflow/c/c_api_experimental.cc

```diff
@@ -743,3 +748,47 @@ void TF_ImportGraphDefOptionsSetValidateColocationConstraints(
     TF_ImportGraphDefOptions* opts, unsigned char enable) {
   opts->opts.validate_colocation_constraints = enable;
 }
+
+// Load a Pluggable Device library
+// On success, returns the handle to library in result and return OK from the
+// function. Otherwise return nullptr in result and error Status from the
+// function.
+//
+// If `library_filename` has already been loaded, we return a cached handle.
+// Device and Kernels/Ops are registered as globals when a library is loaded
+// for the first time.
+TF_Library* TF_LoadPluggableDeviceLibrary(const char* library_filename,
+                                          TF_Status* status) {
+#if defined(IS_MOBILE_PLATFORM) || defined(IS_SLIM_BUILD)
+  status->status = tensorflow::errors::Unimplemented(
+      "PluggableDevice plugin functionality is not supported on mobile");
+  return nullptr;
+#else
+  TF_Library* lib_handle = new TF_Library;
+  static tensorflow::mutex mu(tensorflow::LINKER_INITIALIZED);
```

Comment:
Does this have a non-trivial destructor?  If so, then you should also create this on the heap and leak the object, like `loaded_libs`.

Reply:
I believe [tensorflow::mutex](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/mutex.h) is a traivial destructor. And [LoadDynamicLibrary](https://github.com/tensorflow/tensorflow/blob/f2c163919f51657ec0684bf415b3821af1afdc4f/tensorflow/core/framework/load_library.cc#L46) has simlar code
```
Status LoadDynamicLibrary(const char* library_filename, void** result,
                          const void** buf, size_t* len) {
  static mutex mu(LINKER_INITIALIZED);
  static std::unordered_map<string, Library> loaded_libs;
  Env* env = Env::Default();
```